### PR TITLE
IOS-2856 Fix requesting the gaslimit

### PR DESCRIPTION
**Pull request by tureck1y**
Теперь gaslimit запрашивается правильно, с передачей всех параметров из wc-транзакции. 

Изменения в blockchain-sdk отсюда:
https://github.com/tangem/blockchain-sdk-swift/pull/209

### CODE REVIEW

**Comment by megakoko** [comment]
There are conflicts in the branch

File: Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift [hunk 1]
```diff
@@ -174,8 +178,8 @@ class WalletConnectTransactionHandler: TangemWalletConnectRequestHandler {
         return .justWithError(output: gasPrice)
     }
 
-    private func getGasLimit(for amount: Amount, destination: String, data: String?, txSender: EthereumGasLoader) -> AnyPublisher<Int, Error> {
-        return txSender.getGasLimit(amount: amount, destination: destination)
+    private func getGasLimit(to: String, from: String, value: String?, data: String?, txSender: EthereumGasLoader) -> AnyPublisher<Int, Error> {
```
Comment (megakoko) [inline Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:181]: Shall we rename it to gasLoader?

**Review by megakoko — APPROVED** [review]

File: Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift (hunk 1 above)
Comment (tureck1y) [reply Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:181]: Not exactly. We must distinguish between gasprice and gaslimit requests.

Comment (megakoko) [reply Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:181]: I mean the argument name is `txSender`

**Review by ba1ashov — APPROVED** [review]

File: Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift (hunk 1 above)
Comment (tureck1y) [reply Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:181]: Definitely, yes 😬

File: Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift [hunk 2]
```diff
@@ -159,7 +163,7 @@ class WalletConnectTransactionHandler: TangemWalletConnectRequestHandler {
         }
     }
 
-    private func getGasPrice(for amount: Amount, tx: WalletConnectEthTransaction, txSender: EthereumGasLoader, decimalCount: Int) -> AnyPublisher<Int, Error> {
+    private func getGasPrice(tx: WalletConnectEthTransaction, txSender: EthereumGasLoader, decimalCount: Int) -> AnyPublisher<Int, Error> {
```
Comment (Andoran90) [inline Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:166]: you can remove `decimalCount` from arguments

Comment (tureck1y) [reply Tangem/App/Services/WalletConnectService/WalletConnect/Handlers/TransactionHandler/WalletConnectTransactionHandler.swift:166]: I wanna refactor this method totally in a develop branch

**Review by Andoran90 — APPROVED** [review]

**Pull request merged**